protostar/starknet/cheaters/block_info.py

```diff
@@ -69,3 +69,7 @@ def apply(self, parent: Self) -> None:
             **parent.contract_address_to_block_number,
             **self.contract_address_to_block_number,
         }
+
+
+def replace_in_marshmallow_dataclass(instance: Any, **changes: Any):
```

Comment:
Is this meant to be in this file?

Reply:
Yes, as a temporary solution. I couldn't find a proper place for this function so, I decided to keep it where it is used.  Once this code is removed, this function could be moved to `block_info_controller`. I didn't put it there in this PR, to avoid creating dependency `protostar.starknet -> protostar.cheatable_starknet`.